api/v1alpha1/deploymenttargetclaim_types.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// DeploymentTargetClaimSpec defines the desired state of DeploymentTargetClaim
+type DeploymentTargetClaimSpec struct {
+	DeploymentTargetClassName DeploymentTargetClassName `json:"deploymentTargetClassName"`
+}
+
+type DeploymentTargetClassName string
+
+const (
+	DTCTName_IsolationLevelNamespace DeploymentTargetClassName = "isolation-level-namespace"
```

Comment:
I believe this constant doesn't need to be defined in this package, and I don't think there is anything special about the `isolation-level-namespace` string constant.

`isolation-level-namespace` just happens to be the name of the `DeploymentTargetClass` they use in their example in ADR-8 doc. In practice, one could use any name for the DTClass, as long as it has this field `provisioner: appstudio.redhat.com/devsandbox`